### DISCUSSION TITLE
Acess home network trough VPS

### DISCUSSION THREAD

**Discussion by deadrabbit87** [wg-easy/wg-easy#1518] [Q&A]
Hi, 

my ISP is behind a cgnat. Therefore i want to use a VPS (Virtual Private Server) with wg-easy like this: 

Client <--> VPS <--> home network

I've set up docker compose with nginx sucessfully on the VPS.

My home network is `192.168.40.0/24`, the wireguard network is `10.9.0.0/24`. 

The connection is established and i can reach the IPs inside the network `10.9.0.0/24`, but not the network `192.168.40.0/24`. 

I figured out, that i have to change the `AllowedIPs` of the servers `wg0.conf`. 

This is what wg0.conf looks like inside the container: 
```
# Note: Do not edit this file directly.
# Your changes will be overwritten!


# Server
[Interface]
PrivateKey = removed
Address = 10.9.0.2/24
ListenPort = 51820
PreUp =
PostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -t nat -A POSTROUTING -o %i -j MASQUERADE
PreDown =
PostDown = iptables -D FORWARD -i %i -j ACCEPT; iptables -t nat -D POSTROUTING -o %i -j MASQUERADE


# Client: QNAP (removed)
[Peer]
PublicKey = removed
PresharedKey = removed
AllowedIPs = 10.9.0.2/32/app #
```
What i need is this: 

```
# Server
[Interface]
PrivateKey = removed
Address = 10.9.0.2/24
ListenPort = 51820
PreUp =
PostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -t nat -A POSTROUTING -o %i -j MASQUERADE
PreDown =
PostDown = iptables -D FORWARD -i %i -j ACCEPT; iptables -t nat -D POSTROUTING -o %i -j MASQUERADE


# Client: QNAP (removed)
[Peer]
PublicKey = removed
PresharedKey = removed
AllowedIPs = 10.9.0.2/32, 192.168.40.0/24
```
As i see here https://github.com/wg-easy/wg-easy/blob/94d87681c384689cb4aa371d9d9c3a924f40ce5f/src/lib/WireGuard.js#L89-L125 

that wan't be possible atm, because AllowedIPs of the peers is only set to it's own ip: 
`AllowedIPs = ${client.address}/32`;`

Is this correct, that i can't use wg-easy as i would like to?

**Comment by LedPeach** [comment]
I'm having the same issue with the same setup. Looking forward to some answers. 

Though, I don't think the ```AllowedIPs = ${client.address}/32```  on line 114 affects the routing. It simply presents the virtual IP address of the said client in the config file. If you were to export the config of that said client from the WebUI, the AllowedIPs will be the one you set up in your docker-compose file or docker-run command line. 

One thing I know for certain is that the gateway of your wg0 interface, the wireguard server ```10.9.0.1``` (which it says ```10.9.0.2``` in your config, which I think is a typo), needs to be able to ping the local network IP address ```192.168.40.0/24```. Otherwise, it doesn't know where to route. By that, you would probably need a very complicated iptables command, which I'm not able to put together. 

The solution for me is just to use the virtual IP range ```10.9.0.1/24``` to access all the local machines, by port forwarding the lan interface to the wireguard interface on the local router. That said, I do very much like to use the lan IP range ```192.168.40.0/24``` for accessing the machines on lan.

**Reply by deadrabbit87** [reply]
No, the `10.9.0.2` is no typo. It's due to the resrictions of the peer (a QNAP NAS), in which it is not able to the set the IP adress in the WireGuard exension to another value than `xxx.xxx.xxx.1`.  Therefore my VPS has the `10.9.0.2` and the QNAP Peer the `10.9.0.1`. 

If i use WireGuard without wg-easy, everything works flawless. In this setup my wg0.conf looks like this: 
```
[Interface]
Address = 10.9.0.2/24
ListenPort = 51820
PrivateKey = removed
MTU = 1450
PostUp = iptables -A FORWARD -i %i -j ACCEPT; iptables -t nat -A POSTROUTING -o %i -j MASQUERADE
PostDown = iptables -D FORWARD -i %i -j ACCEPT; iptables -t nat -D POSTROUTING -o %i -j MASQUERADE

## QNAP Peer
[Peer]
PublicKey = removed

AllowedIPs = 10.9.0.1/32,192.168.40.0/24
PersistentKeepalive = 15
```

If i remove the `192.168.40.0/24` on the AllowedIPs of the Peer, i'm not able any more to reach devices inside the `192.168.40.0/24` network. 

I think your solution is not fully clear to me, Could you please explain it in a more detailied way? 

Thanks!

**Reply by LedPeach** [reply]
https://github.com/wg-easy/wg-easy/discussions/300#discussioncomment-3025446

I think you are correct. The fastest way would be to fork the repo and re-build it yourself. Just follow the comment above by editing the line ```AllowedIPs = ${client.address}/32`;``` to ```AllowedIPs = ${client.allowedIPs}`;```  here https://github.com/wg-easy/wg-easy/blob/94d87681c384689cb4aa371d9d9c3a924f40ce5f/src/lib/WireGuard.js#L114

My solution is just a ninja fix to get everything running. Instead of using the ```192.168.40.0/24``` subnet like you wanted, I use the wireguard virtual network's subnet ```10.9.0.2/24``` to access the "home network" in your "Client <--> VPS <--> home network" set up. 
To do that, you would need either A. install wireguard client on every device or B. install wireguard on your home network router and set up port forwarding & static routing on that router.
